### PR TITLE
WIP: Proposed CTS checks for local accessor type alias

### PR DESCRIPTION
**Pull request by mkinsner**
Proposed tests for https://github.com/KhronosGroup/SYCL-Docs/pull/17

### CODE REVIEW

**Comment by mkinsner** [comment]
I have not run this as part of CTS.  @bader or someone else working on CTS, are you able to validate this change please?

**Comment by bader** [comment]
@mkinsner, I'll check.

File: tests/accessor/accessor_api_local_common.h [hunk 1]
```diff
@@ -97,6 +98,23 @@ class check_local_accessor_api_methods {
...
+      /** check local accessor type alias
+      */
+      static_assert(std::is_same<
+          decltype( local_accessor<T,dims>{range, h} ),
+          decltype( accessor<T, dims, mode, target> {range, h} )
+          >::value, "local accessor type alias check");
+
+      /** check default value of dims parameter in local accessor type alias
+      */
+      if (1 == dims) {
```
Comment (bader) [inline tests/accessor/accessor_api_local_common.h]: `static_assert` is compile time check, so this `if` statement do not enable it only for `1 == dims` case.

File: tests/accessor/accessor_api_local_common.h [hunk 2]
```diff
@@ -97,6 +98,23 @@ class check_local_accessor_api_methods {
         check_return_type<explicit_pointer_t<T, target>>(log, acc.get_pointer(),
                                                          "get_pointer()");
       }
+
+      /** check local accessor type alias
+      */
+      static_assert(std::is_same<
+          decltype( local_accessor<T,dims>{range, h} ),
```
Comment (bader) [inline tests/accessor/accessor_api_local_common.h]:
```suggestion
          decltype( cl::sycl::local_accessor<T,dims>{range, h} ),
```

File: tests/accessor/accessor_api_local_common.h [hunk 3]
```diff
@@ -97,6 +98,23 @@ class check_local_accessor_api_methods {
...
+          decltype( accessor<T, dims, mode, target> {range, h} )
```
Comment (bader) [inline tests/accessor/accessor_api_local_common.h]:
```suggestion
          decltype( cl::sycl::accessor<T, dims, mode, target> {range, h} )
```

File: tests/accessor/accessor_api_local_common.h (hunk 2 above)
Comment (bader) [inline tests/accessor/accessor_api_local_common.h]: This constructor is not available for `dims == 0`.

**Comment by fraggamuffin** [comment]
Sept 9:
*  Mike K working on this

**Comment by mkinsner** [comment]
> These changes look good, though in the past we tended to avoid compile-time tests (i.e. `static_assert`) as these can cause the tests to fail at a much higher granularity, and instead have the check performed at runtime so that even if that check fails the rest of the test will continue to execute and you get more fine-grained test output.
> 
> I think this will also resolve the point that @bader raised about the `static_aassert`.

Yes, agreed.  I almost have the rework ready to commit - will do so shortly.

**Comment by keryell** [comment]
Discussed during Tuesday 2020/02/18 informal meeting:
Mike Kinsner will work on it with a low priority.

**Comment by mkinsner** [comment]
Alternate PR coming later.  Closing this one to avoid confusion.

**Pull request closed without merging**